File: src/LIT.Smabu.UseCases/Payments/Update/UpdatePaymentCommand.cs
using LIT.Smabu.Domain.PaymentAggregate;
using LIT.Smabu.UseCases.Shared;

namespace LIT.Smabu.UseCases.Payments.Update
{
    public record UpdatePaymentCommand : ICommand
    {
        public PaymentId PaymentId { get; }
        public string Details { get; }
        public string Payer { get; }
        public string Payee { get; }
        public string ReferenceNr { get; }
        public DateTime? ReferenceDate { get; }
        public DateTime AccountingDate { get; }
        public decimal AmountDue { get; }
        public DateTime? DueDate { get; }
        public PaymentStatus Status { get; }

        public UpdatePaymentCommand(PaymentId paymentId, string details, string payer, string payee, string referenceNr, DateTime? referenceDate, 
            DateTime accountingDate, decimal amountDue, PaymentStatus status)
        {
            PaymentId = paymentId;
            Details = details;
            Payer = payer;
            Payee = payee;
            ReferenceNr = referenceNr;
            ReferenceDate = referenceDate;
            AccountingDate = accountingDate;
            AmountDue = amountDue;
            Status = status;
        }
    }
}

File: src/lit.smabu.react/src/types/domain/update-payment-command.ts
import { PaymentCondition } from './payment-condition';
import { PaymentId } from './payment-id';
import { PaymentMethod } from './payment-method';
import { PaymentStatus } from './payment-status';

export interface UpdatePaymentCommand {
  paymentId?: PaymentId;
  details?: string | null;
  payer?: string | null;
  payee?: string | null;
  referenceNr?: string | null;
  referenceDate?: Date | null;
  accountingDate?: Date;
  amountDue?: number;
  dueDate?: Date | null;
  paymentMethod?: PaymentMethod;
  status?: PaymentStatus;
  paymentCondition?: PaymentCondition;
}
